From ec84b206b373463f92a079948ea154ef4d0328b0 Mon Sep 17 00:00:00 2001
From: Hendrik <git@h3ndrk.de>
Date: Fri, 4 Jun 2021 19:40:09 +0200
Subject: [PATCH] Add find_dependency() to portaudioConfig.cmake.in

---
 cmake/portaudioConfig.cmake.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/portaudioConfig.cmake.in b/cmake/portaudioConfig.cmake.in
index cacc476..5385841 100644
--- a/cmake/portaudioConfig.cmake.in
+++ b/cmake/portaudioConfig.cmake.in
@@ -1 +1,4 @@
+include(CMakeFindDependencyMacro)
+find_dependency(ALSA REQUIRED)
+
 include("${CMAKE_CURRENT_LIST_DIR}/portaudioTargets.cmake")
-- 
2.31.1

