diff --git a/scripts/build/arch/x86.sh b/scripts/build/arch/x86.sh
index 0538d66c..5e3afb3b 100644
--- a/scripts/build/arch/x86.sh
+++ b/scripts/build/arch/x86.sh
@@ -19,7 +19,7 @@ CT_DoArchTupleValues() {
             winchip*)                     CT_TARGET_ARCH=i486;;
             pentium|pentium-mmx|c3*)      CT_TARGET_ARCH=i586;;
             pentiumpro|pentium*|athlon*)  CT_TARGET_ARCH=i686;;
-            core2|atom)                   CT_TARGET_ARCH=i686;;
+            core2|atom|silvermont)        CT_TARGET_ARCH=i686;;
             prescott|nocona)              CT_TARGET_ARCH=i686;;
             k8*|opteron*)                 CT_TARGET_ARCH=i686;;
             *)                            CT_TARGET_ARCH=i586;;
