--- a/sysdeps/x86/Makefile	2015-05-29 16:45:00.820115660 +0200
+++ b/sysdeps/x86/Makefile	2015-05-29 16:55:24.008156172 +0200
@@ -1,6 +1,6 @@
 ifeq ($(subdir),elf)
 CFLAGS-.os += $(if $(filter $(@F),$(patsubst %,%.os,$(all-rtld-routines))),\
-		   -mno-sse -mno-mmx)
+		   -mno-sse -mno-mmx -mfpmath=387)
 
 tests-special += $(objpfx)tst-ld-sse-use.out
 $(objpfx)tst-ld-sse-use.out: ../sysdeps/x86/tst-ld-sse-use.sh $(objpfx)ld.so
