--- ncurses-5.2/include/curses.tail~	Sat Jun 16 13:47:06 2001
+++ ncurses-5.2/include/curses.tail	Thu Mar 21 16:58:13 2002
@@ -111,8 +111,6 @@
 extern NCURSES_EXPORT(int)     NCURSES_SP_NAME(mouseinterval) (SCREEN*, int);                                      │~                                                                                                                      
 #endif
 
-#define mouse_trafo(y,x,to_screen) wmouse_trafo(stdscr,y,x,to_screen)
-
 /* other non-XSI functions */
 
 extern NCURSES_EXPORT(int) mcprint (char *, int);	/* direct data to printer */
